db/view/view.cc

```diff
@@ -784,14 +853,55 @@ void view_updates::update_entry(const partition_key& base_key, const clustering_
         return;
     }
 
-    deletable_row& r = get_view_row(base_key, update);
-    auto marker = compute_row_marker(update);
-    r.apply(marker);
-    r.apply(update.tomb());
+    auto view_rows = get_view_rows(base_key, update, std::nullopt);
+    auto update_marker = compute_row_marker(update);
+    for (const auto& [row, misc] : view_rows) {
+        auto marker = *(std::get_if<row_marker>(&misc) ?: &update_marker);
+        row->apply(marker);
+        row->apply(update.tomb());
 
-    auto diff = update.cells().difference(*_base, column_kind::regular_column, existing.cells());
-    add_cells_to_view(*_base, *_view, std::move(diff), r.cells());
-    _op_count++;
+        auto diff = update.cells().difference(*_base, column_kind::regular_column, existing.cells());
+        add_cells_to_view(*_base, *_view, std::move(diff), row->cells());
+    }
+    _op_count += view_rows.size();
+}
+
+void view_updates::update_entry_for_computed_column(
+        const partition_key& base_key,
+        const clustering_row& update,
+        const std::optional<clustering_row>& existing,
+        gc_clock::time_point now) {
+    if (!matches_view_filter(*_base, _view_info, base_key, *existing, now)) {
+        create_entry(base_key, update, now);
+        return;
+    }
+    if (!matches_view_filter(*_base, _view_info, base_key, update, now)) {
+        do_delete_old_entry(base_key, *existing, update, now);
+        return;
+    }
+
+    /*
+    if (can_skip_view_updates(update, existing)) {
+        return;
+    }
+    */
+
+    struct misc_visitor {
```

Comment:
it seems like this visitor could be made a little more generic and allow you to drop the calls to `get_if<row_marker>` and `get_if<shadowable_tombstone_tag>` above, maybe it's worth considering.

Also, please add a short note on what this structure is, because `misc` is not a very descriptive name

Reply:
I changed the interface, but I'm still not perfectly content with it. Perhaps you have some opinion on the updated code?